src/sidebar/components/test/annotation-test.js

```diff
@@ -730,37 +730,58 @@ describe('annotation', function() {
         fakeAnnotationMapper.flagAnnotation = sandbox.stub();
       });
 
-      it(
-        'calls annotationMapper.flag() when an annotation is flagged',
-        function(done) {
-          var parts = createDirective();
-          fakeAnnotationMapper.flagAnnotation.returns($q.resolve());
-          parts.controller.flag();
-          assert.calledWith(fakeAnnotationMapper.flagAnnotation,
-              parts.annotation);
-          done();
-        }
-      );
+      context('when the user is not logged in', function() {
+        beforeEach(function() {
+          delete fakeSession.state.userid;
```

Comment:
Strictly speaking, the `userid` will be `null` rather than `undefined` if the user is not logged in.